spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -814,7 +821,17 @@ public Object receiveAndConvert() throws AmqpException {
 
 	@Override
 	public Object receiveAndConvert(String queueName) throws AmqpException {
-		Message response = receive(queueName);
+		return receiveAndConvert(queueName, this.receiveTimeout);
+	}
+
+	@Override
+	public Object receiveAndConvert(long timeoutMillis) throws AmqpException {
+		return receiveAndConvert(this.getRequiredQueue(), timeoutMillis);
+	}
+
+	@Override
+	public Object receiveAndConvert(String queueName, long timeoutMillis) throws AmqpException {
+		Message response = receive(queueName, timeoutMillis);
```

Comment:
Failing tests - need to call `doReceiveNoWait` if the timeout is 0 - will fix on merge.